src/app/services/guard.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable, takeUntil } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GuardService implements CanActivate {
```

Comment:
This should be something like LoginGuard instead of GuardService, and can live in a guards folder. Otherwise, looks good!